centrallix-sysdoc/OSDriver_Authoring.md

```diff
@@ -115,497 +170,1032 @@ License:  Copyright (C) 2001-2011 LightSys Technology Services. See LICENSE.txt
+    /** Build the new filename. **/
+    count = snprintf(obj->Pathname->Pathbuf, 256, "%s/%s", qy->Parent->Obj->Pathname->Pathbuf, new_name);
+    if (count < 0 || 256 <= count) return NULL;
+    obj->Pathname->Elements[obj->Pathname->nElements++] = strrchr(obj->Pathname->Pathbuf, '/') + 1;
+```
+
+### Function: QueryCreate()
+```c
+void* xxxQueryCreate(void* qy_v, pObject new_obj, char* name, int mode, int permission_mask, pObjTrxTree *oxt);
+```
+<!-- TODO: Greg - I guessed this object signature from multiquery.  However, the variable names appear to be inconsistent with similar functions, and I'm overall not very confident about this. -->
+**No documentation provided.**
+
 
-QueryDelete is passed the qy_v void* parameter, and an oxt parameter. It should return 0 on successful deletion, and -1 on failure.
+### Function: QueryClose()
+```c
+int xxxQueryClose(void* qy_v, pObjTrxTree* oxt);
+```
+The close function closes a query instance, freeing all allocated data and releasing all shared memory such as open connections, files, or other driver instances.  This function operates very similarly to `Close()`, documented in detail above.  The query should be closed, whether or not `QueryFetch()` has been called enough times to enumerate all of the query results.
 
-QueryClose is also passed qy_v and oxt.  It should close the query, whether or not QueryFetch has been called enough times to enumerate all of the query results.
 
-### F.  Managing Object Attributes
-All objects will have at least some attributes.  Five attributes are mandatory: 'name', 'content_type', 'inner_type', 'outer_type', and 'annotation'.  All compliant drivers must implement these five attributes, all of which have a data type of DATA_T_STRING.
+### Object Attributes
+All objects can have attributes, and there are five required attributes that all drivers must implement (explained below).
 
 Currently, the OS specification includes support for the following data types:
 
-- DATA_T_INTEGER	-	32-bit signed integer.
-- DATA_T_STRING	-	Zero-terminated ASCII string.
-- DATA_T_DOUBLE	-	Double-precision floating point.
-- DATA_T_DATETIME	-	date/time structure.
-- DATA_T_MONEY	-	money data type.
+| Name              | Description
+| ----------------- | ------------
+| `DATA_T_INTEGER`  | 32-bit signed integer.
+| `DATA_T_STRING`   | Null-terminated ASCII string.
+| `DATA_T_DOUBLE`   | Double-precision floating point number.
+| `DATA_T_DATETIME` | Date/time structure.
+| `DATA_T_MONEY`    | Money structure.
+
+See `datatypes.h` for more information.
+
+For `true`/`false` or `on`/`off` attributes, use `DATA_T_INTEGER` where 0 indicates `false` and 1 indicates `true`.
+
+The following five attributes are required (all are of type `DATA_T_STRING`):
+
+| Attribute    | Description
+| ------------ | ------------
+| name         | The name of the object, just as it appears in any directory listing.  The name of the object must always be unique for its directory.
```

Comment:
e.g. the primary key of a database row or name of a file in a directory.

Reply:
Oops, this wasn't general enough. Updated.